scripts/tests/yaml/tests_finder.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env -S python3 -B
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os.path
+import sys
+
+import click
+
+_JSON_FILE_EXTENSION = '.json'
+_YAML_FILE_EXTENSION = '.yaml'
+_KNOWN_PREFIX = 'Test_TC_'
+
+_KEYWORD_ALL_TESTS = 'all'
+_DEFAULT_DIRECTORY = 'src/app/tests/suites/'
+
+_CI_CONFIGURATION_NAME = 'ciTests'
+_MANUAL_CONFIGURATION_NAME = 'manualTests'
+
+
+class TestsFinder:
+    def __init__(self, configuration_directory: str = _DEFAULT_DIRECTORY, configuration_name: str = _CI_CONFIGURATION_NAME):
+        self.__test_directory = _DEFAULT_DIRECTORY
+        self.__test_collections = self.__get_collections(configuration_directory, configuration_name)
+
+    def get_default_directory() -> str:
```

Comment:
Is `get_default_directory` used anymore? If not we can remove it

Reply:
Removed.